content/sips/sip-311.md

```diff
@@ -53,31 +53,52 @@ Teleporters involve adding `transferCrossChain` and `ccipReceive` functions to t
+
+**Enhanced security via Rate Limits**
+
+In contrast to arbitrary messaging, CCIP token transfers provide configurable rate limits per token pool per lane, which are set up in alignment with the token issuer. This feature is part of the heavily audited CCIP code base and only available for token transfers. CCIP token transfers also benefit from increased security via the global rate limits (across token pools), so not every token’s rate limit could be exploited at maximum in a worst case scenario.
+
+#### Fee Collection
 
 This proposal entails collecting fees in ETH, rather than snxUSD, for the following reasons:
 
 - Users will always receive the same amount of snxUSD that they have transferred.
 - This is the most efficient method to collect fees, as no additional token transfers are involved in the transaction.
 - Users are already paying ETH for gas fees to execute the transaction.
+- Reduces code complexity since the `CcipRouter` handles the fee calculation for the execution of the specified message.
+- Users are automatically refunded any left over ETH that was unused in the message execution.
+
+In this scenario, there is also potential to set up a rev-share model between CCIP and Synthetix (as token issuer) for third-party CCIP snxUSD token transfers, creating a shared incentive for increased snxUSD token transfer volume over CCIP.
 
 Future SIPs could involve additional fee payment methods (such as snxUSD), alternative fee structures, and circuit-breaking functionality for added security.
 
-### Technical Specification
+### Implementation
 
 <!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones.-->
 
-Implementation entails two functions on the snxUSD token module:
+Implementation entails adding two new functions to the snxUSD token module:
 
-#### `transferCrossChain(uint chainId, address recipient, uint amount) payable returns (bool)`
+**`transferCrossChain(uint64 destChainId, address to, uint256 amount) external payable returns (uint256 gasTokenUsed)`**
 
 - Confirm that the sender has a sufficient balance of snxUSD to transfer.
-- Determine the fee amount in ETH by taking an SCCP-specified percentage of the amount transferred (plus any SCCP-specified flat fee) and denominate that amount in ETH based on a price oracle.
-- Transfer the fee into the contract and notify a rewards distributor of the collected fees.
-- Encode the `recipient` and `amount` and call `ccipSend`.
+- Encode the `destChainId`, `to` and `amount` with selector `mint` and call `ccipSend`.
 - Burn the amount of snxUSD transferred.
+- Refunds any left over gas token (i.e. ETH).
```

Comment:
Suggest to clarify that this refund is specifically for the difference between the value that the user passes to transferCrossChain() and what CCIP Router returns as getFee(). See also comment above under 'Fee Collection'